src/System.Net.Http/src/System/Net/Http/Managed/HttpConnectionPool.cs

```diff
@@ -583,9 +577,34 @@ public CachedConnection(HttpConnection connection)
             /// terminate it, which would be considered a failure, so this race condition is largely benign and inherent to
             /// the nature of connection pooling.
             /// </returns>
-            public bool IsUsable(DateTimeOffset now) =>
-                now - _returnedTime <= TimeSpan.FromMilliseconds(MaxIdleTimeMilliseconds) &&
-                IsUsable();
+            public bool IsUsable(DateTimeOffset now)
```

Comment:
This is a bit of a nit, but I wonder if the code would be better structured if we changed this to take timeout and idleTimeout parameters instead of fetching the HttpConnectionSettings every time.  This is always called by the pool, right?  And the pool can retrieve the timeout settings once and then pass to this method as needed.

Reply:
Ok